src/components/Base/Form/form.jsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import styles from './index.scss';
+
+export default class Form extends React.Component {
+  static propTypes = {
+    className: PropTypes.string,
+    onSubmit: PropTypes.func,
+  }
+
+  static defaultProps = {
+    className: '',
+  }
+
+  constructor(props) {
+    super(props);
+    this._formData = props.data || {};
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if ('data' in nextProps) {
+      this._formData = nextProps.data || {};
```

Comment:
since `this._formData` already has a default value in constructor, no need to assign `{}` here.

Reply:
`data` in `nextProps ` may not be an object, should validate it firstly

Reply:
`data` should be defined in propTypes

Reply:
ok